### DISCUSSION TITLE
How to call table.on_select?

### DISCUSSION THREAD

**Discussion by ed2050** [zauberzeug/nicegui#2968] [Q&A]
Either basic documentation on event handling is missing, or it needs to be indexed better.  I can't find anything useful on `table.on_select`.

The extent of [the docs on it](https://nicegui.io/documentation/table) say this:
```py
on_select(callback: Callable[..., Any]) -> Self
    Add a callback to be invoked when the selection changes.
```

That is supremely unhelpful.  What the heck arguments does the callback take?  Any is not descriptive at all.  What kind of object am I given?  The table row?  Some type of event object?  What properties does it have?

I tried searching the forums and docs with no luck.  The most I found is a [discussion of how to create a new table cell event](https://github.com/zauberzeug/nicegui/discussions/979), which doesn't reveal how on_select works.  Even google didn't turn up anything.

The doc page on generic events doesn't help either.  [This one](https://nicegui.io/documentation/generic_events).  It says `The generic event handler can ... optionally takes GenericEventArguments as argument`.  So what the heck is `GenericEventArguments` and where is it documented??

One last note - the generic events page gives two urls for more info on events, and both are plain text.  Surely they should be hyperlinks.

Anyone have an example with table.on_select?

**Comment by falkoschindler** [comment]
Hi @ed2050,

> How to call table.on_select?

Usually you don't. But if you can't use `ui.table`'s `on_select` parameter, the `on_select` method is an alternative. These methods have been introduced in PR #2708 as an extension to event handler arguments that are passed to the initializer.

It's true that the type annotation of such callbacks can be improved. When we introduced type annotation in the code base, we just used `Callable[..., Any]` to please mypy and to deal with it later. But now that the code base is type safe in most other places, it might be time to fix these callbacks. Would you like to create a pull request?
